docs/content/rust/guides/roles/getting-started.md

```diff
@@ -0,0 +1,274 @@
+- `"org"` is the identifier for this resource type (this can be any string
+  you choose).
+- `actions` is a list enumerating all the actions that may be performed on the
+  resource.
+- `roles` is a dictionary defining all the roles for this resource.
+
+In our rule body, we first define the list of available actions for this
+resource:
+
+```polar
+resource(_type: Org, "org", actions, roles) if
+    actions = ["read", "create_repo"] and
+    roles = {
+        ...
+    };
+```
+
+Now, we define our roles. Roles are defined in a dictionary that maps
+the role name to a role configuration.
+
+```polar
+resource(_type: Org, "org", actions, roles) if
+    actions = ["read", "create_repo"] and
+    roles = {
+        member: {
+            permissions: ["read"],
+        },
+        owner: {
+            permissions: ["read", "create_repo"],
+        }
+    };
+```
+
+This resource definition defines two roles:
+
+- _member_: Has the `read` permission.
+- _owner_: Has the `read` and `create_repo` permissions.
+
+Permissions are actions associated with a resource type. A permission can
+directly reference an action defined in the same resource. Later, we'll
+see how to leverage relationships between resources to grant a role a
+permission defined on a different resource
+
+{{% callout "resource(...) is just a rule" "blue" %}}
+
+The `resource` definition is just a regular Polar rule. That's why it
+has an `if` and `and` between variable assignments. `actions` and
+`roles` are unbound parameters, meaning they can be assigned inside of
+the rule body.
+
+We could have written this rule without a body:
+
+```polar
+resource( _type: Org, "org", ["read", "create_repo"],
+    {
+        member: {
+            permissions: ["read"],
+        },
+        owner: {
+            permissions: ["read", "create_repo"],
+        }
+    }
+);
+```
+
+But we think the expanded form is clearer.
+
+{{% /callout %}}
+
+### Adding `role_allows` to our policy
+
+To allow access based on roles, we add the following `allow` rule
+
+```polar
+allow(actor, action, resource) if
+    # add other conditions can be added here if needed
+    role_allows(actor, action, resource);
+```
+
+Oso will now allow access to any resource that is allowed based on the
+role definitions. 
+
+### Assigning roles to users
+
+Now we've configured roles and set up our policy. For users to have
+access, we must assign them roles.
+
+You can use your own data models for roles with Oso. You just need to tell us
+what roles a user has for a particular resource through the
+`actor_has_role_for_resource` rule. As an example, our types might have an
+interface like this:
+
+```rust
+pub struct OrgRole {
+  pub name: String,
+  pub resource: Org,
+}
+
+impl User {
+  pub fn roles(&self) -> Vec<OrgRole> {
+    // ...
+  }
+}
+```
+
+And the `actor_has_role_for_resource` would be implemented as:
+
+```polar
+actor_has_role_for_resource(actor, role_name, resource) if
+    role in actor.roles and
```

Comment:
Does this need parentheses? Like `role in actor.roles()`? It's possible it doesn't, but just want to make sure